### PR TITLE
Support latest EKS versions

### PR DESCRIPTION
**Pull request by keyvaann**
This PR adds support for latest version of EKS and its addons. I did a test install for v1.31 and installation was successful. I assume v1.30 will work as wel.

### CODE REVIEW

File: cluster/data.tf [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
+        "coredns"        = "v1.11.3-eksbuild.2"
+        "kube_proxy"     = "v1.29.10-eksbuild.3"
+        "vpc_cni"        = "v1.19.0-eksbuild.1"
+        "ebs_csi_driver" = "v1.37.0-eksbuild.1"
       }
     },
-    "1.26" : {
-      "cluster_version" = "1.26"
+    "1.28" : {
+      "cluster_version" = "1.31"
```
Comment (baixiac) [inline cluster/data.tf:32]: I can see the mismatch between keys and cluster_versions in the EKS version dict. Are they typos or intentional?

File: README.md [hunk 2]
```diff
@@ -122,6 +122,8 @@ Created resources (if all enabled):
 
 Make sure to install [terraform-docs](https://github.com/terraform-docs/terraform-docs) and run `make prepare` before making a commit to make sure the documentation is up to date and the code is valid.
 
+In order to support new version of EKS you need to make sure the addons that we use are compatible with the new target version. You can get a list of addons and their EKS compatiblity with running `aws eks describe-addons-versions` and then searching for the addons that are defined in `cluster/data.tf`.
+
```
Comment (baixiac) [inline README.md:126]: 👍

File: cluster/data.tf (hunk 1 above)
Comment (keyvaann) [reply cluster/data.tf]: They're fixed now.

**Review by baixiac — APPROVED** [review]
LGTM

**Pull request merged**